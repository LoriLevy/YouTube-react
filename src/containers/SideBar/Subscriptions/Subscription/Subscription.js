
import React from 'react';
import { Icon, Image, Menu } from "semantic-ui-react";
import './Subscription.scss';

export function Subscription(props) {

    let rightEle, emt = null;
    const { broadcasting, amountNewVideos } = props;
    if (broadcasting) {
        rightElement = <Icon name='signal' />;
    }
    else if (amountNewVideos) {
        rightElement = <span className='new-videos-count'>{amountNewVideos}</span>
    }

    return (
        <Menu.Item>
            <div className='subscription'>
                <div>
                    {/* For now use placeholder for the channel icon. 
                    We havenâ€™t connected our app to the Youtube backend yet. */}
                    <Image src='http://via.placeholder.com/28x28' avatar />
                    <span>{props.label}</span>
                </div>
                {rightElement}
            </div>
        </Menu.Item>
    );
}