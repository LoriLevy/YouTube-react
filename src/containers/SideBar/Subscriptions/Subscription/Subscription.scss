@import '../../../../styles/shared.scss';


/*  For layout purposes we wrap the content on the left into a
 .subscription div element that serves as a flex container. 
 We force the flex container to maximise the space between 
 its elements by setting justify-content: space-between. 
 The align-items: center property centers the div content vertically. */
.subscription {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;

    i.icon {
        color: $red;
    }

    .new-videos-count {
        color: $grey;
    }
}