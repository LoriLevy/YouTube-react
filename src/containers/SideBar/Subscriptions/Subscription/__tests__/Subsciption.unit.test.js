import React from 'react';
import { shallow } from 'enzyme';
import { Subscription } from '../Subscription';

describe('Subscription', () => {
    test("Empty subscriptions renders", () => {
        const wrapper = shallow(
            <Subscription />
        );
        expect(wrapper).toMatchSnapshot();
    });
});