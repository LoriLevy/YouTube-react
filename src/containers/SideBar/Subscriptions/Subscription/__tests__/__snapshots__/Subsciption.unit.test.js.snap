// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription Empty subscription renders 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;

exports[`Subscription broadcasting Subscription renders  1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Loritest
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`Subscription non-broadcasting Subscription with new videos renders  1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Loritest
      </span>
    </div>
    <span
      className="new-videos-count"
    >
      3
    </span>
  </div>
</MenuItem>
`;
